GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace GenericsHomework
+{
+    public class Node<T> : ICollection<T>
+    {
+        private T _Value;
+        private Node<T> _Next;
+
+        public Node(T value)
+        {
+            _Value = value;
+            _Next = this;
+        }
+
+        public T Value
+        {
+            get
+            {
+                return _Value;
+            }
+
+            private set
+            {
+                _Value = value;
+            }
+
+        }
+
+        public Node<T> Next
+        {
+            get
+            {
+                return _Next;
+            }
+
+            private set
+            {
+                value._Next = _Next;
+                _Next = value;
+            }
+
+        }
+
+        public override string ToString()
+        {
+            return _Value?.ToString() ?? "";
+        }
+
+        public Node<T> Insert(T value)
+        {
+            Node<T> node = new(value);
+            Next = node;
+
+            return node;
+        }
+
+        public void Clear()
+        {
+            // If there are no other references to other nodes in the list then GC will eventually cleanup the inaccessible nodes.
+            _Next = this;
+        }
+
+        public int Count
+        {
+            get
+            {
+                int count = 1;
+                Node<T> next = this.Next;
+
+                while (next != this)
+                {
+                    count++;
+                    next = next.Next;
+                }
+
+                return count;
+            }
+        }
+
+
+        public bool IsReadOnly => false;
+
+        public void Add(T value)
+        {
+            Node<T> head = this;
+            Node<T> current = this;
+
+            while (current.Next != head)
+            {
+                current = current.Next;
+            }
+
+            current.Insert(value);
+        }
+
+        public bool Contains(T value)
+        {
+            Node<T> head = this;
+            Node<T> current = this;
+
+            do
+            {
+                if ((current.Value == null && value == null) || (current.Value != null && current.Value.Equals(value)))
```

Comment:
So the question here is whether you want a reference equality check against null, or if you want the [potentially] overridden equality operator (if one is defined) on `T`. To get reference equality you can do:
```suggestion
                if ((current.Value is null && value is null) || (current.Value is not null && current.Value.Equals(value)))
```

Similar comments on the other methods that have similar lines of code.